File: app/utils/response.py
# ---------------------------------------------------------------------------
# response.py — Standardised JSON envelope helpers
# ---------------------------------------------------------------------------
from flask import jsonify


def success_response(data=None, message: str = None, status_code: int = 200):
    """Return a standard success envelope.

    Shape:
        { "success": true, "data": {...}, "message": "optional string" }
    """
    payload = {"success": True, "data": data}
    if message:
        payload["message"] = message
    return jsonify(payload), status_code


def paginated_response(data: list, page: int, limit: int, total: int, status_code: int = 200):
    """Return a standard paginated success envelope.

    Shape:
        {
          "success": true,
          "data": [...],
          "pagination": { "page": 1, "limit": 20, "total": 84, "hasNext": true }
        }
    """
    payload = {
        "success": True,
        "data": data,
        "pagination": {
            "page": page,
            "limit": limit,
            "total": total,
            "hasNext": (page * limit) < total,
        },
    }
    return jsonify(payload), status_code


def error_response(
    code: str,
    message: str,
    field: str = None,
    status_code: int = 400,
):
    """Return a standard error envelope.

    Shape:
        { "success": false, "error": { "code": "...", "message": "...", "field": null } }
    """
    payload = {
        "success": False,
        "error": {
            "code": code,
            "message": message,
            "field": field,
        },
    }
    return jsonify(payload), status_code

File: src/utils/response.ts
// src/utils/response.ts — Consistent API response helpers
import type { Response } from 'express';

export function success(res: Response, data: any, status = 200) {
  return res.status(status).json({ data, status: 'success' });
}

export function paginated(
  res: Response,
  data: any[],
  pagination: { page: number; limit: number; total: number; hasNext: boolean },
) {
  return res.json({ data, pagination, status: 'success' });
}

export function error(
  res: Response,
  message: string,
  status = 400,
  code?: string,
  field?: string,
) {
  return res.status(status).json({
    error: { code: code ?? 'ERROR', message, ...(field ? { field } : {}) },
  });
}
